src/EncryptedFS.ts

```diff
@@ -3228,7 +3281,11 @@ class EncryptedFS {
     let nextDir, nextPath, target, targetType;
     await this.iNodeMgr.transact(
       async (tran) => {
-        target = await this.iNodeMgr.dirGetEntry(tran, curdir, parse.segment);
+        if (parse.segment === '..' && curdir === this.rootIno) {
```

Comment:
Can you explain how this works?

Reply:
This seems to rely on the idea that the `curdir` would be always set to `this.rootIno`.

But the `curdir` could be changed to other locations. That's the point of the `curdir`.

Can you add a test for what happens when the `curdir` is changes to other locations?

Reply:
If the current directory inode index is equal to the root iNode that is set at construction and the current parsed segment is going up a directory (..) then it will set the next target as the current directory (not moving up a directory)

Reply:
otherwise it will go up a directory as normal. This hasn't affected any of the other tests

Reply:
But the `curdir` index can be changed. It's not something that is static.

Reply:
the navigate function will iterate through dirs

Reply:
Oh I must have been confused. The `curdir` here is not the `CurrentDirectory`, it's local to this navigation recursion?

Reply:
yep correct

Reply:
At any case you should have a test to see what happens if the `CurrentDirectory` were to be changed. So prevent users from changing current directory above the chroot.

Also what happens if the current directory is inside another chroot.

One more test for nesting chroots. What if you chroot, and then chroot again.

Reply:
When all those pass, then you should be able to merge.

Do you have the npm permissions to publish?

Reply:
Remember:

```
npm patch minor
npm run build
git push
git push --tags
npm publish --access public
```

Reply:
Updated the above commands.

Reply:
Yeah I should have the right permissions